storage/backup/backup-cli/src/metadata/view.rs

```diff
@@ -127,6 +128,47 @@ impl MetadataView {
 
         Ok(res)
     }
+
+    /// Compact the epoch ending metdata files and merge compaction_cnt files into 1 metadata file
+    /// The generated chunks should be sorted based on version
+    fn compact_backups<T>(backups: &[T], compaction_cnt: usize) -> Result<Vec<&[T]>> {
+        // Initialize an empty vector to store the output
+        let mut output_vec = Vec::new();
+
+        // Iterate over the input vector in chunks of three
+        for chunk in backups.chunks(compaction_cnt) {
```

Comment:
hmm.. we probably need something more sophisticated, because:
1. rerunning the compaction on top of a previous run, the result will be larger and larger files
2. and because of newer files showing up, the result files will be of different sizes in terms of number of original backups contained
3. we are replacing all files every time, which seems not too efficient?

Reply:
1. it would still be the same file since the chunk logic is based on the in-memory Vec<Metadata>.
2. as 1
3. we can save the progress on the compaction only do compaction and replace files after a version, epoch. But not sure about the priority. The total metadata size is still less than 5M.  It could become a problem once it is a 100x larger.

Reply:
added